### DISCUSSION TITLE
Error processing request when trying to click SSO login, URL schemes?

### DISCUSSION THREAD

**Discussion by pandaboy6621** [9p4/jellyfin-plugin-sso#218] [Q&A] [answered]
Hi guys. I am trying to set up this plugin with authentik and am getting an Error processing request when I try to login.
I connect to jellyfin with https://jellyfin.local.[redacted].net/ which is behind a traefik reverse proxy. 

My URLs in authentik and Jellyfin are https://jellyfin.local.[redacted].net/sso/OID/start/authentik 
and strict: https://jellyfin.local.[redacted].net/sso/OID/redirect/authentik 

My authentik version is [2024.10.4]

Here are some logs that I found in Jellyfin:

[2024-12-14 23:04:57.490 -08:00] [INF] [24] Jellyfin.Plugin.SSO_Auth.Api.SSOController: SSO Controller initialized
[2024-12-14 23:04:58.637 -08:00] [ERR] [24] Jellyfin.Api.Middleware.ExceptionMiddleware: Error processing request. URL "GET" "/sso/OID/start/authentik".

**Comment by 9p4** [comment]
What kind of SSL certificate are you using?

**Comment by pandaboy6621** [comment]
It's a cert from traefik that is generated using lets encrypt and a
Cloudflare domain.

**Reply by 9p4** [reply]
Is Cloudflare proxying the requests (orange cloud or grey cloud in the CF dashboard?)

**Comment by pandaboy6621** [comment]
I think grey, the url I have only works when I'm using my local pi hole as
dns, that Jellyfin url is not remotely accessible.

**Reply by 9p4** [reply]
Are you using a Jellyfin container?

**Comment by pandaboy6621** [comment]
Yes but it is configured as an external service in traefik because that
container is on a different device.

**Reply by 9p4** [reply]
Can you run `docker exec -it jellyfin /bin/cat /etc/resolv.conf`, replacing `jellyfin` with the name of your Jellyfin container (can be seen with `docker ps`)?

**Comment by pandaboy6621** [comment]
```# Generated by Docker Engine.
# This file can be edited; Docker Engine will not make further changes once it
# has been modified.

nameserver 127.0.0.11
options ndots:0

# Based on host file: '/etc/resolv.conf' (internal resolver)
# ExtServers: [1.1.1.1 8.8.8.8 host(2001:4860:4860::8888)]
# Overrides: []
# Option ndots from: internal```

**Comment by 9p4** [comment]
Can you please upload the entire Jellyfin log file and your config once you make a request that fails?

**Reply by pandaboy6621** [reply]
Here are the logs: https://pastebin.com/CBRA69nS
(I notice that its looking for https)

OID Endpoint = http://XXX.XXX.X.XX:7000/application/o/jellyfin/

**Reply by 9p4** [reply]
Is your authentik server behind traefik/https?

**Reply by pandaboy6621** [reply]
It is not, should I set it up that way?

**Reply by 9p4** [reply]
Yes

**Reply by pandaboy6621** [reply]
Authentik runs on the same server as Traefik so I can apply labels to the authentik-server container right?

**Reply by 9p4** [reply]
I'm not sure, I don't know how you have your system set up. I just know that you should be running your OpenID system behind HTTPS.

**Reply by pandaboy6621** [reply]
Ok I can now get to authentik at https://sso.local.[redacted].net/, So I just need to change the URL in the plugin settings now?

**Reply by 9p4** [reply]
Yes

**Reply by pandaboy6621** [reply]
I set the OID endpoint to https://sso.local.[redacted].net/application/o/jellyfin/.well-known/openid-configuration and refreshed the login page and I still get the same login issue. Does Jellyfin need to be restarted every time a change is made to the plugin settings?

**Reply by 9p4** [reply]
Set it to the URL without a `.well-known/openid-configuration` at the end of it. Jellyfin does not need to be restarted upon a config change.

**Reply by pandaboy6621** [reply]
I set it to https://sso.local.[redacted].net/application/o/jellyfin/ and still have the same error

**Reply by 9p4** [reply]
Can you please upload the log file again?

**Reply by pandaboy6621** [reply]
Here: https://pastebin.com/D4Fz6qBw

**Reply by 9p4** [reply]
You say that your SSO domain is set via a local DNS server? Can you set that nameserver in the Jellyfin container's /etc/resolv.conf with `docker exec -it jellyfin echo nameserver 127.0.0.1 ">>" /etc/resolv.conf`, replacing the IP with your DNS server's IP?

**Reply by pandaboy6621** [reply]
Ok let me try that

**Reply by 9p4** [reply, marked as answer]
To be clear, this is not an issue with the plugin, but is instead an issue with your setup: the Jellyfin container is misconfigured and cannot resolve your Authentik server's DNS name.

**Reply by pandaboy6621** [reply]
I tried docker exec -it 212df0812179 echo nameserver 192.168.1.22 ">>" /etc/resolv.conf and then ran the cat command, It looks like it added my internal DNS at the .94 address # ExtServers: [192.168.1.94 8.8.8.8 host(2001:4860:4860::8888)] but my internal one I still the 127 one

**Reply by 9p4** [reply]
Can you create a shell in the Jellyfin container and install `dig` (run the exec command with `bash`, then `apt install dnsutils`), and run `dig [yourauthentikdomain]`?

**Comment by pandaboy6621** [comment]
It looks like it is working now, should I still try and replace that 127
address?

**Reply by 9p4** [reply]
That fix was temporary, and once the Jellyfin container restarts, it will not work. You shouldn't replace any other addresses. You should look into adding a DNS server to a Docker container. As I don't personally use Docker, I can't really help any further.

**Comment by pandaboy6621** [comment]
I have an issue with my login now when I go to login with SSO it makes a user called Authentik Admins and 3a82b63653bde925b5b3639f313b005afc471d06bb3788d11d83384bd359d6a5, how do I configure my default user group that I made in Autentik (authentik general users) and (Authentik Admins) as Jellyfin Admins?

**Reply by pandaboy6621** [reply]
I figured that out, I had to set it to users username, however my groups are still not applying, I don't understand how to set the scopes in Authentik or the role fields in the plugin itself. Right now I made the Group scope mapping from [the docs](https://github.com/9p4/jellyfin-plugin-sso/blob/main/providers.md#authentik) And I have those two user groups (mentioned above) each with one user in those groups.
